src/tests/src/study/system-model/test_component.cpp

```diff
@@ -0,0 +1,213 @@
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/study/system-model/component.h"
+
+#include "utils.h"
+
+using namespace Antares::Study::SystemModel;
+
+struct ComponentTestFixture
+{
+    static Model createModel(bool with_parameters);
+};
+
+Model ComponentTestFixture::createModel(bool with_parameters)
+{
+    ModelBuilder model_builder;
+    model_builder.withId("model");
+    if (with_parameters)
+    {
+        model_builder.withParameters({Parameter("param1",
+                                                ValueType::FLOAT,
+                                                Parameter::TimeDependent::NO,
+                                                Parameter::ScenarioDependent::NO),
+                                      Parameter("param2",
+                                                ValueType::FLOAT,
+                                                Parameter::TimeDependent::NO,
+                                                Parameter::ScenarioDependent::NO)});
+    }
+    auto model = model_builder.build();
+    return model;
+}
+
+BOOST_AUTO_TEST_SUITE(_Component_)
+
+BOOST_FIXTURE_TEST_CASE(nominal_build_with_parameters, ComponentTestFixture)
+{
+    Model model = createModel(true);
+    ComponentBuilder component_builder;
+    auto component = component_builder.withId("component")
+                       .withModel(&model)
+                       .withParameterValues({{"param1", 5}, {"param2", 3}})
+                       .withScenarioGroupId("scenario_group")
+                       .build();
+    BOOST_CHECK_EQUAL(component.Id(), "component");
+    BOOST_CHECK_EQUAL(component.getModel(), &model);
+    BOOST_CHECK_EQUAL(component.getParameterValue("param1"), 5);
+    BOOST_CHECK_EQUAL(component.getParameterValue("param2"), 3);
+    BOOST_CHECK_EXCEPTION(component.getParameterValue("param3"),
+                          std::invalid_argument,
+                          checkMessage("Parameter 'param3' not found in component 'component'"));
+    BOOST_CHECK_EQUAL(component.getScenarioGroupId(), "scenario_group");
+}
+
+BOOST_FIXTURE_TEST_CASE(nominal_build_without_parameters1, ComponentTestFixture)
+{
+    Model model = createModel(false);
+    ComponentBuilder component_builder;
+    auto component = component_builder.withId("component2")
+                       .withModel(&model)
+                       .withParameterValues({})
```

Comment:
About` .withParameterValues({})` : do we need that ? 
I mean, by default a component has an empty list of parameters, right ?
There are 2 occurrences of it in these tests.

Reply:
no we don't need it ; this is to test that the behavior of calling withParameterValues with an empty map is equivalent to when we don't call it at all